ros/angel_system_nodes/angel_system_nodes/task_monitor_v2.py

```diff
@@ -250,15 +282,18 @@ def thread_run_hmm(self):
                     self._skip_score = self._hmm.get_skip_score()
                 log.info(f"HMM computation time: {time.time() - start_time}")
 
-                if self._current_step != step:
+                step_id = self._hmm.model.class_str.index(step)
+
+                # Only change steps if we have a new step and it is not background
+                if self._current_step != step and step_id != 0:
                     self._previous_step = self._current_step
                     self._current_step = step
```

Comment:
Is this current formulation, are we reporting 1-indexed steps now? That might be at odds with the natural usage downstream, i.e. we may want to convert to zero-indexed values going into the message we output in order to be congruent with the task graph we output via the service also exposed here.